File: frontend/src/types/character.ts
export interface Character {
  id: string;
  world_id: string;
  real_user_id?: string;
  is_ai: boolean;
  display_name: string;
  avatar_media_id?: string;
  avatar_url?: string;
  meta?: string;
  created_at: string;
  role?: string;
}

export interface CharacterListResponse {
  characters: Character[];
}

File: services/character-service/internal/models/character.go
package models

import (
	"database/sql"
	"encoding/json"
	"time"
)

type Character struct {
	ID            string
	WorldID       string
	RealUserID    sql.NullString // NULL for AI characters
	IsAI          bool
	DisplayName   string
	AvatarMediaID sql.NullString
	Meta          json.RawMessage
	CreatedAt     time.Time
}

type CreateCharacterParams struct {
	WorldID       string
	RealUserID    sql.NullString
	DisplayName   string
	AvatarMediaID sql.NullString
	Meta          json.RawMessage
}